### DISCUSSION TITLE
Trying to compile/build project but i get many errors since this project got migrated from .net 4.8 to .net 6

### DISCUSSION THREAD

**Discussion by ladnene** [VorlonCD/bi-aidetection#347] [Q&A] [answered]
Hi everyone,
I want to contribute to this project and i have downloaded multiple versions (commits) of the project which are from the commit "[Update BUILD.bat](https://github.com/gentlepumpkin/bi-aidetection/commit/f6f168b7db877bdd01d769128fa7802904ad7a30)" and before and i could build them and run the app.
But since this commit [* Migrated from .net 4.8 to .net 6. As a result the UI is a bit faste…](https://github.com/gentlepumpkin/bi-aidetection/commit/9763735a7ae0224df8a37ccadddaf253ddaa35ff) any commit i download up to the master commit i get error saying "
Severity	Code	Description	Project	File	Line	Suppression State	Details
Warning		The referenced project '..\..\..\ObjectListView.NET6\ObjectListView\ObjectListView2022.csproj' does not exist.	UI	C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	2146		

Error	NU1104	Unable to find project 'C:\Users\ladne\Desktop\ObjectListView.NET6\ObjectListView\ObjectListView2022.csproj'. Check that the project reference is valid and that the project file exists.	UI	C:\Users\ladne\Desktop\bi-aidetection-3e87ba0eed59872ea3c0b68dec07fe8b7a44abc2\src\UI\UI.csproj	1"

I tried to find this folder project on internet which i did and i put on the desktop next to the master commit project and try to build it and then i get not more errors in the "Error list" but i get a pop up with this error : 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/b1ee3b35-3e2e-46bc-af2f-503aa0258130)

The debug executable
'C:\Users\ladne\Desktop\bi-aidetection-3e87ba0eed59872ea3c0b68dec07fe8b7a44abc2\src\UI\bin\Debug\net8.0-windows10.0.19041.0\AITool.exe' specified in the 'UI.NET6' debug profile does not exist.

Not sure if the ObjectListView2022 i found in internet is the right one. Can you please provide this project and help fix the pop up error ?

@VorlonCD  Please help me build the latest commit version of this project and thank you very much

**Comment by VorlonCD** [comment]
I think you should use git to download everything.   Try this:

1. Make sure visual studio is fully up to date - Help > Check for updates   (I use 2022 v17.10) but *maybe* older versions work too)
2. Help > Get tools and updates > make sure 'Git for windows' and '.net 8.0' is installed.  Maybe other things needed, not sure
3. Open visual studio > "Clone a repository"
4. Enter https://github.com/VorlonCD/bi-aidetection.git and wait for full download
5. Solution explorer > Solution BI-detection.net6 > right click > Restore nuget packages
6. Solution explorer > Solution bi-detection.net6 > right click > build solution

@tinbum1 used to compile this himself all the time, not sure if has recently.   Maybe there is something else to it I'm missing because I've always had the bits and pieces in place and havent tried from scratch for a while.

**Reply by ladnene** [reply]
@VorlonCD Thank you for the quick reply and clear steps but i tried to follow all of them and it didn't work, here is a summary :

1-updated visual studio from 17.8 to 17.10.1. Durring the update i got this note : Installation done with warnings 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/f9f0eac9-ca9f-4b07-a91a-1e5081b1dacc)
not sure if these warnings have an effect on this project or not but i kept having the same old errors when trying to build the project so don't think they are the ones causing the problem
2-i checked and found the .net 8.0 installed but the git for windows was not installed so i installed it and now i have both : 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/730ca004-df13-4273-8b01-10ef6f62a8de)
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/dd852472-b181-4d4e-a4c6-ef6078db2d21)
3-i did "Clone a repository" for " https://github.com/VorlonCD/bi-aidetection.git" but i got this error :

> Microsoft Visual Studio
> Git failed with a fatal error. early EOF
> fetch-pack: invalid index-pack output

So i tried to clone it with cmd and "git clone  https://github.com/VorlonCD/bi-aidetection.git" , didn't work also so did "git clone --recursive https://github.com/VorlonCD/bi-aidetection.git", didn't work so tried to install Gitkraken, Github Desktop and clone with them but didn't work. then i used the cmd and Increased the Git Buffer Size "git config --global http.postBuffer 1073741824" which also didn't work. Finally the only thing that worked is that i used Git Bash with SSH instead of HTTPS to clone the project.
which cloned the project successfully : 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/d7657fcb-bc8d-40ec-979c-bf2c6271c6ce)
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/abf65b9f-b0d1-4a85-a358-86978f0f7c2c)

4-Opened the "bi-aidetection.NET6.sln"
5-In the solution explorer i got this :  
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/7491f7c7-fb9b-4d1a-baa4-f18e68774dc4)

ObjectListView2022 (not found)

So i looked for this project name and found this repo : https://github.com/geomatics-io/ObjectListView
Which i downloaded using "Download ZIP" and unzipped it in the same folder where the Git Bash cloned the [bi-aidetection] project then i renamed the unzipped folder from "ObjectListView-main" to "ObjectListView.NET6" then reopened the bi-aidetection and now i no longer have missing ObjectListView2022  as you can see : 

![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/ca428d83-9806-4880-97d5-24c9696179ff)

6-i did "Solution explorer > Solution BI-detection.net6 > right click > Restore nuget packages" and got : 

All packages are already installed and there is nothing to restore.
Time Elapsed: 00:00:00.0051570
========== Finished ==========

7-i did "Solution explorer > Solution bi-detection.net6 > right click > build solution" and i got this output : 

Build started at 23:54...
1>------ Skipped Build: Project: AITool.Service ------
1> 
2>------ Skipped Build: Project: UI ------
2> 
========== Build: 0 succeeded, 0 failed, 1 up-to-date, 2 skipped ==========
========== Build completed at 23:54 and took 00,064 seconds ==========

As you can see AITool.Service and UI builds are skipping and i think this is what giving me this error when i try to debug the project : 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/32c26d1a-c614-442e-9553-addf56132a97)

![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/fabc630b-b19a-4a28-a600-325e8dbf7a8c)

Maybe the because the build is skipping the UI project so there is no AITool.exe generated. why is it skipping thought ? i am getting no errors. i don't understand what i am doing wrong

@VorlonCD please help

**Comment by ladnene** [comment, marked as answer]
It finally worked !

Once I removed the UI and AITool.Service project from the solution and added it back, it is no longer skipped.

Then during build i got this error : 

"Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	CS0006	Metadata file 'C:\Users\ladne\ObjectListView.NET6\ObjectListView\bin\Release\ObjectListView.dll' could not be found	UI	C:\Users\ladne\bi-aidetection\src\UI\CSC	1"

So i tried to remove the ObjectListView from the solution and try to build : 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/4a91e2b4-88a1-4141-b7bd-1e9f02957463)

And finally it worked ! yay ! 
![image](https://github.com/VorlonCD/bi-aidetection/assets/35897057/97d03f33-1bbc-4b15-af98-c7b1cc8a011d)

@VorlonCD thank you very much for your support

**Reply by VorlonCD** [reply]
Great!   "Aitool.Service" and "ThreadsafeTesting" are not required for it to work, but the copy of ObjectListView I used had to be modified to support .NET 8 is required.   

I guess since the copy I use is in not in the same repository and may have caused your original issue?    You can download this and add it to the project.

https://github.com/VorlonCD/ObjectListView.NET6

It may be working because the .exe found an existing copy of ObjectListview.  For example: "C:\Tools\AITool\ObjectListView.dll"

**Reply by ladnene** [reply]
Yes this was it, i had to use your modified version of ObjectListView 

Here what i did for a fresh and clean build : 
1- I downloaded bi-aidetection repository using "Download ZIP" in github repo page and unzipped it in a folder 
2-I downloaded https://github.com/VorlonCD/ObjectListView.NET6 repository using "Download ZIP" in github repo page and unzipped it in the same folder and renamed the folder from "ObjectListView.NET6-main" to "ObjectListView.NET6"
3-Solution explorer > Solution BI-detection.net6 > right click > Restore nuget packages
4-Solution explorer > Solution bi-detection.net6 > right click > build solution

And everything worked fine 💯 
@VorlonCD  Thank you for providing the https://github.com/VorlonCD/ObjectListView.NET6 repo !
